mteb/evaluation/MTEB.py

```diff
@@ -322,6 +325,8 @@ def run(
 
         Args:
             model: Model to be used for evaluation
+            task_to_prompt_name: Dictionary mapping task names to prompt names. If None, the default prompt names are used.
+            model_prompts: Dictionary mapping task types to prompt names. If None, the default prompt names are used.
```

Comment:
Shouldn't this be specified in the model? I don't think we should specify it here.

Similar with `task_to_prompt_name`.

Reply:
There's an issue where, if users pass a `SentenceTransformer`, there won't be a mapping from tasks to prompts. I wanted to allow specifying this, but I can remove that option if needed.

Reply:
Can you give an example of this?

Reply:
In the README example:

```python
import mteb
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
tasks = mteb.get_tasks(tasks=["Banking77Classification"])
evaluation = mteb.MTEB(tasks=tasks)
results = evaluation.run(model, output_folder=f"results/{model_name}")
```
It's not possible to add prompts to the model in this way. However, this may be acceptable since users should create custom models instead.

Reply:
They can specify prompts like this I believe:

```
model = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2", prompts ={"task_name": "myprompt"}
```

Reply:
Yes, this approach could remove `model_prompts`, but how would the mapping between tasks and `prompt_name` be specified? Should I just keep `task_to_prompt_name`? I don't want to do that because some models in SentenceTransformers already have prompts and I wanted to reuse them

Reply:
I would image that the user would specify the prompt for a given task like:

```
model = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2", 
    prompts={
      # task specific:
      "BornholmBitextMining": "Retrieve parallel sentences in Danish and Bornholmsk",
      # retrieval with prompt-type:
      "MIRACLReranking-query": "Query: Given a question, retrieve Wikipedia passages that answer the question",
      "MIRACLReranking-corpus": "", # or just leave out
    }
)
```

Let me know if there is something I am missing?

Reply:
I can merge task_to_prompt_name and model prompts and check only model prompts to make this work like this